model_zoo/models/molecular_autoencoder_candle_pilot2/model_conv_molecular_autoencoder_pilot2.prototext

```diff
@@ -1,92 +1,85 @@
 model {
-  name: "sequential_model"
-  data_layout: "data_parallel"
   mini_batch_size: 1024
   block_size: 256
   num_epochs: 4
   num_parallel_readers: 1
   procs_per_model: 0
 
-  ###################################################
+  ##############################################
   # Objective function
-  ###################################################
+  ##############################################
 
   objective_function {
-    mean_squared_error {}
-    mean_absolute_deviation {
+    layer_term { layer: "mean_squared_error" }
+    layer_term {
+      layer: "mean_absolute_error"
       scale_factor: 0.01
     }
     l2_weight_regularization {
       scale_factor: 0.0005
     }
   }
 
-  ###################################################
+  ##############################################
   # Callbacks
-  ###################################################
-  callback {
-    print {
-      interval: 1
-    }
-  }
-  callback {
-    timer {
-    }
-  }
-  # callback {
-  #   save_images {
-  #     image_prefix: "images_"
-  #     image_format: "pgm"
-  #   }
-  # }
+  ##############################################
 
-  ###################################################
-  # start of layers
-  ###################################################
+  callback { print {} }
+  callback { timer {} }
+
+  ##############################################
+  # Layers
+  ##############################################
+
+  ######################
+  # Data
+  ######################
 
-  #######
-  # INPUT
-  #######
   layer {
-    name: "data"
+    name: "input"
+    children: "data label"
     data_layout: "data_parallel"
     input {
       io_buffer: "partitioned"
     }
   }
-
-  #######
-  # Flatten the data to 1d
-  ######
   layer {
-    name: "reshape1"
+    parents: "input"
+    name: "data"
     data_layout: "data_parallel"
-    num_neurons_from_data_reader: true
     reshape {
-      reshape_to_flattened_conv_format: true
+      dims: "1 -1" # Reshape to 1xX tensor
     }
   }
+  layer {
+    parents: "input"
+    name: "label"
+    data_layout: "data_parallel"
+    dummy {}
+  }
 
-  #############
-  # CONVOLUTION 1
-  #############
+  ######################
+  # Encoder
+  ######################
+
+  # encode1
   layer {
-    name: "mol_conv1"
+    parents: "data"
+    name: "encode1_conv"
     data_layout: "data_parallel"
     convolution {
       num_dims: 1
       num_output_channels: 1024
       conv_dims: "240"
       conv_pads: "0"
       conv_strides: "240"
-      weight_initialization: "he_normal"
-      has_bias: true
+      has_bias: false
```

Comment:
Is there a reason that you disabled the bias for several of the layers?

Reply:
They're followed by batch norm, so the bias will just be removed.

Reply:
I see what you are saying and so from a technical point of view I agree.  For this refactoring, I would like to keep things as a literal translation without changing the behavior.